### DISCUSSION TITLE
Page content disappears on page reload

### DISCUSSION THREAD

**Discussion by e-e** [adonisjs/core#2896] [Help]
Currently happening with both Chrome (Version 91.0.4472.114) and Firefox (89.0.2), MacOS. 

When refreshing a page, or clicking a link that would load the same path that the page is currently on, when the page "loads", the content disappears.

Refreshing works using Safari, but clicking the link shown below downloads an empty file called test.

https://user-images.githubusercontent.com/5191838/125174702-36695880-e17c-11eb-83f9-e3361b14820a.mov

Doing a hard refresh (cmd + shift + r) reloads with the content.

Relevant files:

```typescript
// start/routes.ts

Route.get("/test", "TestController.index");
```

```typescript
// app/Controllers/Http/TestController.ts

import { HttpContextContract } from "@ioc:Adonis/Core/HttpContext";

export default class TestController {
  public async index({ view }: HttpContextContract) {
    return view.render("test/index");
  }
}
```

```html
// resources/views/test/index.edge
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <title>Document</title>
</head>
<body>
  This is just a test

<hr>

<a href="/test">test link to same page</a>

</body>
</html>
```

```json
{
  "name": "_",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "build": "node ace build --production",
    "start": "node server.js",
    "dev": "node ace serve --watch",
    "lint": "eslint . --ext=.ts",
    "format": "prettier --write .",
    "tests": "node -r @adonisjs/assembler/build/register japaFile.ts"
  },
  "devDependencies": {
    "@adonisjs/assembler": "^5.3.2",
    "adonis-preset-ts": "^1.1.0",
    "eslint": "^7.17.0",
    "eslint-config-prettier": "^7.1.0",
    "eslint-plugin-adonis": "^1.0.15",
    "eslint-plugin-prettier": "^3.3.1",
    "execa": "^5.0.0",
    "get-port": "^5.1.1",
    "japa": "^3.1.1",
    "pino-pretty": "^4.3.0",
    "prettier": "^2.2.1",
    "typescript": "^4.1.3",
    "youch": "^2.1.1",
    "youch-terminal": "^1.0.1"
  },
  "dependencies": {
    "@adonisjs/auth": "^8.0.7",
    "@adonisjs/core": "^5.1.9",
    "@adonisjs/lucid": "^15.0.3",
    "@adonisjs/mail": "^7.2.2",
    "@adonisjs/redis": "^5.0.9",
    "@adonisjs/repl": "^3.1.4",
    "@adonisjs/session": "^6.1.0",
    "@adonisjs/shield": "^7.0.4",
    "@adonisjs/view": "^6.0.4",
    "axios": "^0.21.1",
    "luxon": "^1.25.0",
    "pg": "^8.5.1",
    "phc-bcrypt": "^1.0.4",
    "proxy-addr": "^2.0.6",
    "reflect-metadata": "^0.1.13",
    "source-map-support": "^0.5.19",
    "supermemo": "^2.0.17"
  }
}
```

Also, I recently followed the steps with the project to upgrade from v5 preview to the release

Update: On the requests where the page is blank, the server as actually returning a completely empty response
Update 2: Actually, it doesn't matter if a link is referring to the current page or not - following a link will go to that path but no content is actually returned

**Comment by thetutlage** [comment]
Can you open the dev tools and look/share the response body and headers?

**Reply by e-e** [reply]
Yep, here is the hard refresh vs the soft refresh. Hard refresh response body is the normal html document you would expect, soft refresh is completely empty

[hard-refresh.txt](https://github.com/adonisjs/core/files/6804198/hard-refresh.txt)
[soft-refresh.txt](https://github.com/adonisjs/core/files/6804200/soft-refresh.txt)

quick side-by-side comparison:
<img width="1792" alt="Screen Shot 2021-07-12 at 1 24 58 PM" src="https://user-images.githubusercontent.com/5191838/125351139-909d2180-e314-11eb-8a55-05428605f095.png">

**Reply by thetutlage** [reply]
That means the server is sending the empty content. Can you put your inside a repo and then share it with me, so that I can reproduce the issue? Just try to keep it minimal and focused on the actual issue :)

**Reply by e-e** [reply]
@thetutlage https://github.com/e-e/adonis-empty-response

**Reply by thetutlage** [reply]
Already. Seems like the issue when etags are enabled. The server should be returning a 304 status, whereas it returns 200.

**Comment by thetutlage** [comment]
Reported here https://github.com/adonisjs/http-server/issues/45

**Reply by e-e** [reply]
Thank you for looking into this. I have confirmed on my end also that disabling the etag app config indeed fixes the issue. I'll keep that disabled for now and watch the above issue for a resolution.
